iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -254,6 +263,34 @@ public void commitAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable
     }
   }
 
+  @Override
+  public void rollbackAlterTable(org.apache.hadoop.hive.metastore.api.Table hmsTable, EnvironmentContext context)
+      throws MetaException {
+    if (Boolean.valueOf(context.getProperties().getOrDefault(MIGRATE_HIVE_TO_ICEBERG, "false"))) {
+      LOG.debug("Initiating rollback for table {} at location {}",
+          hmsTable.getTableName(), hmsTable.getSd().getLocation());
+      context.getProperties().put(INITIALIZE_ROLLBACK_MIGRATION, "true");
+      this.catalogProperties = getCatalogProperties(hmsTable);
+      try {
+        this.icebergTable = Catalogs.loadTable(conf, catalogProperties);
+      } catch (NoSuchTableException nte) {
+        // iceberg table was not yet created, no need to delete the metadata dir separately
+        return;
+      }
+
+      // we want to keep the data files but get rid of the metadata directory
+      String metadataLocation = ((BaseTable) this.icebergTable).operations().current().metadataFileLocation();
```

Comment:
I wanted to put a comment here, but not sure if I have done it or forget it 😄
Could we create some `CatalogUtil.dropTableMetaData(deleteIo, deleteMetadata);` based on `CatalogUtil.dropTableData(deleteIo, deleteMetadata);`?

Reply:
As we discussed offline, we will leave the metadata dir delete as it is.